src/Nethermind/Nethermind.Core/Extensions/Bytes.cs

```diff
@@ -1140,5 +1141,91 @@ public static void ChangeEndianness8(Span<byte> bytes)
                     (BinaryPrimitives.ReverseEndianness(endIth), BinaryPrimitives.ReverseEndianness(ith));
             }
         }
+
+        public static string ToCleanUtf8String(this byte[] bytes)
+        {
+            // The maximum number of UTF-16 chars is bytes.Length, but each Rune can be up to 2 chars.
+            // So we allocate bytes.Length to bytes.Length * 2 chars.
+            const int maxOutputChars = 32 * 2;
+
+            if (bytes == null || bytes.Length == 0 || bytes.Length > 32)
+                return string.Empty;
+
+            // Allocate a char buffer on the stack.
+            Span<char> outputBuffer = stackalloc char[maxOutputChars];
+
+            int outputPos = 0;
+            int index = 0;
+            bool hasValidContent = false;
+            bool shouldAddSpace = false;
+
+            while (index < bytes.Length)
+            {
+                ReadOnlySpan<byte> span = bytes.AsSpan(index);
+
+                OperationStatus status = Rune.DecodeFromUtf8(span, out Rune rune, out var bytesConsumed);
+                if (status == OperationStatus.Done)
+                {
+                    if (!IsControlCharacter(rune))
+                    {
+                        if (shouldAddSpace)
+                        {
+                            outputBuffer[outputPos++] = ' ';
+                            shouldAddSpace = false;
+                        }
+
+                        int charsNeeded = rune.Utf16SequenceLength;
+                        if (outputPos + charsNeeded > outputBuffer.Length)
+                        {
+                            // Expand output buffer
+                            int newSize = outputBuffer.Length * 2;
+                            char[] newBuffer = new char[newSize];
+                            outputBuffer.Slice(0, outputPos).CopyTo(newBuffer);
+                            outputBuffer = newBuffer;
+                        }
+
+                        rune.EncodeToUtf16(outputBuffer.Slice(outputPos));
+                        outputPos += charsNeeded;
+                        hasValidContent = true;
+                    }
+                    else
+                    {
+                        // Control character encountered; set flag to add space if needed
+                        if (hasValidContent)
+                            shouldAddSpace = true;
+                    }
+                    index += bytesConsumed;
+                }
+                else if (status == OperationStatus.InvalidData)
```

Comment:
This is same as `else`, so can be just removed?

Reply:
Removed